### ISSUE TITLE
Trove API invalidly returns two access-control-allow-origin headers

### ISSUE THREAD

**Issue by Conal-Tuohy** [Conal-Tuohy/TroveProxy#24]
If a request is made to the Trove API containing an "Origin" request header, then the response includes _two_ `access-control-allow-origin` headers, both with the value `*`. If a request is made _without_ an `Origin` header, then a single `access-control-allow-origin` header is returned. However, requests to the API from a JS client in a browser will always have an `Origin` header, and because [multiple Access-Control-Allow-Origin headers are not allowed](https://developer.mozilla.org/en-US/docs/Web/HTTP/CORS/Errors/CORSMultipleAllowOriginNotAllowed), these requests will fail, making it impossible to call the Trove API from such a client, except by going through a proxy which can remove one of the supernumerary headers. 

This is a Trove server error.

**Comment by Conal-Tuohy** [comment]
Trove's response to the following request contains a single `access-control-allow-origin` header:
```bash
curl --verbose  'https://api.trove.nla.gov.au/v3/result?category=newspaper&n=0&encoding=json'
```
Trove's response to the following request contains _two_ `access-control-allow-origin` headers:
```bash
curl -H 'Origin: http://localhost' --verbose  'https://api.trove.nla.gov.au/v3/result?category=newspaper&n=0&encoding=json'
```

**Comment by Conal-Tuohy** [comment]
Libraries Australia reference: `RSref165736`

**Comment by Conal-Tuohy** [comment]
I've had no confirmation from NLA, but the bug appears to be fixed:

```bash
curl -s --verbose  'https://api.trove.nla.gov.au/v3/result?category=newspaper&n=0&encoding=json' 2>&1 \
| grep --ignore-case --fixed-strings "Access-Control-Allow-Origin:"
```
produces `< access-control-allow-origin: *`

**Comment by wragge** [comment]
Still seems to be broken?

```shell
curl H 'Origin: http://localhost' --verbose  'https://api.trove.nla.gov.au/v3/result?category=newspaper&n=0&encoding=json' 2>&1 | grep --ignore-case --fixed-strings "Access-Control-Allow-Origin:"
```

Produces:

```
< access-control-allow-origin: *
< access-control-allow-origin: *
```

**Comment by Conal-Tuohy** [comment]
You're absolutely right @wragge that this bug persists. My comment from Dec 20 was mistaken because I failed to send an `Origin` header in that test. BTW your example did the same for me, until I realised that `H` should have been `-H`. I will chase it up with the NLA
